containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -874,25 +874,13 @@ class UnorderedMap {
     if (m_hash_lists.data() != src.m_hash_lists.data()) {
       Kokkos::deep_copy(m_available_indexes, src.m_available_indexes);
 
-      using raw_deep_copy =
-          Kokkos::Impl::DeepCopy<typename device_type::memory_space,
-                                 typename SDevice::memory_space>;
-
-      raw_deep_copy(m_hash_lists.data(), src.m_hash_lists.data(),
-                    sizeof(size_type) * src.m_hash_lists.extent(0));
-      raw_deep_copy(m_next_index.data(), src.m_next_index.data(),
-                    sizeof(size_type) * src.m_next_index.extent(0));
-      raw_deep_copy(m_keys.data(), src.m_keys.data(),
-                    sizeof(key_type) * src.m_keys.extent(0));
+      Kokkos::deep_copy(m_hash_lists, src.m_hash_lists);
+      Kokkos::deep_copy(m_next_index, src.m_next_index);
+      Kokkos::deep_copy(m_keys, src.m_keys);
       if (!is_set) {
-        raw_deep_copy(m_values.data(), src.m_values.data(),
-                      sizeof(impl_value_type) * src.m_values.extent(0));
+        Kokkos::deep_copy(m_values, src.m_values);
       }
-      raw_deep_copy(m_scalars.data(), src.m_scalars.data(),
-                    sizeof(int) * num_scalars);
-
-      Kokkos::fence(
-          "Kokkos::UnorderedMap::deep_copy_view: fence after copy to dst.");
+      Kokkos::deep_copy(m_scalars, src.m_scalars);
```

Comment:
Note that we now have up to 12 global fences where 2 would be sufficient.

Reply:
That's a lot indeed. Is it possible to call `deep_copy` without fencing?

Reply:
You could at least use a three-argument deep_copy for all those calls except the first and the last.

Reply:
> That's a lot indeed. Is it possible to call deep_copy without fencing?

That'd what the three-argument `deep_copy` is for.

Reply:
Ok, I think I found a way.